.github/workflows/deploy-to-test.yml

```diff
@@ -0,0 +1,47 @@
+name: Deploy to Test Environment
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  VAULT_PW: ${{ secrets.VAULT_PW }}
+
+permissions:
+    id-token: write   # This is required for requesting the JWT
+    contents: read    # This is required for actions/checkout
+
+jobs:
+  deploy-to-test:
+    name: Deploy to Test Environment
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Decrypt secrets
+        run: |
+          echo $VAULT_PW > .vault_password
+          bash ops/secrets --decrypt
+          mv -fv shared_files/encrypted/* shared_files/decrypted/
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: configure aws credentials
+        uses: aws-actions/configure-aws-credentials@v3
+        with:
+          role-to-assume: arn:aws:iam::${{ secrets.AWS_ACCOUNT_NUMBER }}:role/github-to-aws-oidc
+          aws-region: us-east-1
+
+      - name: Deploy to Test Environment
+        run: |
+          # Stop the existing containers if any
+          docker-compose down
```

Comment:
Should be in cleanup if needed (maybe not given short-lived self-hosted, or definitely not if on github hosted infra)